src/util/class.h

```diff
@@ -1,7 +1,9 @@
 #pragma once
 
-// A macro to disallow the copy constructor and operator= functions
-// This should be used in the private: declarations for a class
+// DEPRECATED: Instead rely on "the rule of zero". See
+// https://en.cppreference.com/w/cpp/language/rule_of_three#Rule_of_zero
+// See also: Cpp Core Guidelines C.20, C.21 & C.22 as well as the clang-tidy rule
+// `modernize-use-equals-delete`
```

Comment:
I don't undestand the comment. The `modernize-use-equals-delete` explict cites this solution as good. 
You are probably refering the old impmentation of the macro. 
```suggestion
// DEPRECATED: Instead delete functions explicit without using this macro. 
```

Reply:
well what I actually want to say that people should reconsider using this macro as per the rule of zero. There are two problems:
1. Its used all over the place (more than is necessary nor good).
2. It violates [CppCoreGL C.21: If you define or =delete any copy, move, or destructor function, define or =delete them all](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c21-if-you-define-or-delete-any-copy-move-or-destructor-function-define-or-delete-them-all). 

So the best practice IMO is that if a class deals with ownership it should be its sole purpose (I think we both agree on that). If it does that, it should adhere to the rule of zero and do that explicitly (not via this macro or by mixing this macro). Classes that don't deal with ownership should rely on implicitly generated SMF's and not delete them explicitly (for example by using this macro). Considering all that, this macro should not exist in code following best practices and thus I mark it deprecated. 

The actual point however is not do it explicitly everywhere, but rather reconsider refactoring to follow the rule of zero, which is why I added all the references. So I don't agree with your suggested change because it does not link to the proper best practices.

> The modernize-use-equals-delete explict cites this solution as good.

I wanted to mention it so people aren't tempted to do it the pre C++11 way.

Reply:
OK, But than write an explicit advice what to do instead of this macro. A short for of your last comment?

Reply:
Sure, I'll try my best.

Reply:
done.